arangod/Indexes/Index.cpp

```diff
@@ -415,6 +415,22 @@ bool Index::Compare(VPackSlice const& lhs, VPackSlice const& rhs) {
         return false;
       }
     }
+  } else if (type == IndexType::TRI_IDX_TYPE_IRESEARCH_LINK) {
+    value = lhs.get("view");
+    if (value.isString()) {
+      if (arangodb::basics::VelocyPackHelper::compare(
+              value, rhs.get("view"), false) != 0) {
+        auto ls = value.copyString();
+        auto rs = rhs.get("view").copyString();
+        if (ls.size() > rs.size()) {
+          std::swap(ls, rs);
+        }
+        if (ls.back() != '/' ||
```

Comment:
it is theoretically possible that though `ls` is a string, it may be the empty string. Then accessing `back()` is UB.

From https://en.cppreference.com/w/cpp/string/basic_string/back:
> Returns reference to the last character in the string. The behavior is undefined if empty() == true. 

This probably is an edge case that will not happen in reality, I am just highlighting it here because I don't know where the "view" values originate from and if they are validated properly before.